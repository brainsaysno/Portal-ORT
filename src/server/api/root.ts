import { createTRPCRouter } from "./trpc";
import { materiaRouter } from "./routers/example";

/**
 * This is the primary router for your server.
 *
 * All routers added in /api/routers should be manually added here
 */
export const appRouter = createTRPCRouter({
  materia: materiaRouter,
});

// export type definition of API
export type AppRouter = typeof appRouter;
