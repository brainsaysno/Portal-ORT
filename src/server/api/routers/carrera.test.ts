import { type inferProcedureInput } from "@trpc/server";
import { describe, expect, test } from "vitest";
import { type AppRouter, appRouter } from "../root";
import { createTRPCContext } from "../trpc";

describe("Carrera router", () => {
  test("byId", async () => {
    const ctx = createTRPCContext();
    const caller = appRouter.createCaller(ctx);

    type Input = inferProcedureInput<AppRouter["carrera"]["getById"]>;
    const input: Input = {
      id: 2485,
    };

    const ingSistemas = await caller.carrera.getById(input);

    expect(ingSistemas).toMatchSnapshot();
  });
});
