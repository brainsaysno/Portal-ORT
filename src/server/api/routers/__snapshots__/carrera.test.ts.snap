// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carrera router > byId 1`] = `
{
  "id": 2485,
  "materias": [
    {
      "id": 1410,
    },
    {
      "id": 1479,
    },
    {
      "id": 1743,
    },
    {
      "id": 1774,
    },
    {
      "id": 1778,
    },
    {
      "id": 1780,
    },
    {
      "id": 1781,
    },
    {
      "id": 1786,
    },
    {
      "id": 1793,
    },
    {
      "id": 1798,
    },
    {
      "id": 1875,
    },
    {
      "id": 2103,
    },
    {
      "id": 2106,
    },
    {
      "id": 3437,
    },
    {
      "id": 3831,
    },
    {
      "id": 3835,
    },
    {
      "id": 3837,
    },
    {
      "id": 3838,
    },
    {
      "id": 3839,
    },
    {
      "id": 3842,
    },
    {
      "id": 3844,
    },
    {
      "id": 3846,
    },
    {
      "id": 3851,
    },
    {
      "id": 3856,
    },
    {
      "id": 3872,
    },
    {
      "id": 3876,
    },
    {
      "id": 3904,
    },
    {
      "id": 3924,
    },
    {
      "id": 3935,
    },
    {
      "id": 4231,
    },
    {
      "id": 4266,
    },
    {
      "id": 4784,
    },
    {
      "id": 4872,
    },
    {
      "id": 4891,
    },
    {
      "id": 4892,
    },
    {
      "id": 4918,
    },
    {
      "id": 5128,
    },
    {
      "id": 5330,
    },
    {
      "id": 5440,
    },
    {
      "id": 5505,
    },
    {
      "id": 5512,
    },
    {
      "id": 5520,
    },
    {
      "id": 5527,
    },
    {
      "id": 5564,
    },
    {
      "id": 5634,
    },
    {
      "id": 5636,
    },
    {
      "id": 5640,
    },
    {
      "id": 5641,
    },
    {
      "id": 5716,
    },
    {
      "id": 5733,
    },
    {
      "id": 5735,
    },
    {
      "id": 5906,
    },
    {
      "id": 5907,
    },
    {
      "id": 6005,
    },
    {
      "id": 6011,
    },
    {
      "id": 6030,
    },
    {
      "id": 6106,
    },
    {
      "id": 6107,
    },
    {
      "id": 6110,
    },
    {
      "id": 6147,
    },
    {
      "id": 6343,
    },
    {
      "id": 6406,
    },
    {
      "id": 6409,
    },
    {
      "id": 6411,
    },
    {
      "id": 6414,
    },
    {
      "id": 6415,
    },
    {
      "id": 6417,
    },
    {
      "id": 6435,
    },
    {
      "id": 6449,
    },
    {
      "id": 6452,
    },
    {
      "id": 6455,
    },
    {
      "id": 6498,
    },
    {
      "id": 6543,
    },
    {
      "id": 6563,
    },
    {
      "id": 6580,
    },
    {
      "id": 6627,
    },
    {
      "id": 6668,
    },
    {
      "id": 6852,
    },
    {
      "id": 6872,
    },
    {
      "id": 6876,
    },
    {
      "id": 6877,
    },
    {
      "id": 6933,
    },
    {
      "id": 6934,
    },
    {
      "id": 7107,
    },
    {
      "id": 7144,
    },
    {
      "id": 7270,
    },
    {
      "id": 7271,
    },
    {
      "id": 7349,
    },
    {
      "id": 7358,
    },
    {
      "id": 7466,
    },
    {
      "id": 7472,
    },
    {
      "id": 7473,
    },
    {
      "id": 7512,
    },
    {
      "id": 7655,
    },
    {
      "id": 7657,
    },
    {
      "id": 7658,
    },
    {
      "id": 7659,
    },
    {
      "id": 7660,
    },
    {
      "id": 7661,
    },
    {
      "id": 7662,
    },
    {
      "id": 7663,
    },
    {
      "id": 7664,
    },
    {
      "id": 7665,
    },
    {
      "id": 7666,
    },
    {
      "id": 7667,
    },
    {
      "id": 7668,
    },
    {
      "id": 7669,
    },
    {
      "id": 7670,
    },
    {
      "id": 7673,
    },
    {
      "id": 7674,
    },
    {
      "id": 7675,
    },
    {
      "id": 7676,
    },
    {
      "id": 7677,
    },
    {
      "id": 7678,
    },
    {
      "id": 7679,
    },
    {
      "id": 7680,
    },
    {
      "id": 7681,
    },
    {
      "id": 7682,
    },
    {
      "id": 7683,
    },
    {
      "id": 7684,
    },
    {
      "id": 7685,
    },
    {
      "id": 7686,
    },
    {
      "id": 7687,
    },
    {
      "id": 7688,
    },
    {
      "id": 7689,
    },
    {
      "id": 7690,
    },
    {
      "id": 7691,
    },
    {
      "id": 7692,
    },
    {
      "id": 7715,
    },
    {
      "id": 7716,
    },
    {
      "id": 7812,
    },
    {
      "id": 7825,
    },
    {
      "id": 7886,
    },
    {
      "id": 7902,
    },
    {
      "id": 8058,
    },
    {
      "id": 8063,
    },
    {
      "id": 8068,
    },
    {
      "id": 8069,
    },
    {
      "id": 8082,
    },
    {
      "id": 8083,
    },
    {
      "id": 8084,
    },
    {
      "id": 8085,
    },
    {
      "id": 8149,
    },
    {
      "id": 8419,
    },
    {
      "id": 8420,
    },
    {
      "id": 8454,
    },
    {
      "id": 8455,
    },
    {
      "id": 8468,
    },
  ],
  "nombre": "IngenierÃ­a en Sistemas",
}
`;
